### ISSUE TITLE
Unable to get HMR working in Vue with TypeScript

### ISSUE THREAD

**Issue by trajano** [NativeScript/NativeScript#9507]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 8.0.2
 - Cross-platform modules:
 - Android Runtime: ^8.0.0
 - iOS Runtime: ^8.0.0
 - XCode Version: 12.5.1
 - Plugin(s):

**Describe the bug**

Trying to get a HMR working with TypeScript and Vue

**To Reproduce**
```
ns create --vue --ts  ns-vue-app
# I also tested with ns create --vue   ns-vue-app no typescript and get similar results
cd ns-vue-app
ns run ios
# app is now in simulator
# modify app/components/Home.vue so the message is changed
#save
```

**Expected behavior**

The modified message appears in the simulator

**Sample project**

**Additional context**

Running on macOS and Simulator is running iPad Air simulator

**Comment by trajano** [comment]
Code that I changed

```
<script lang="ts">
  import Vue from "nativescript-vue";

  export default Vue.extend({
    computed: {
      message() {
        return "Blank {N}-Vue app abcde"; // added abcde
      }
    }
  });
</script>
```

**Comment by trajano** [comment]
ns debug ios sort of works but it does a "restart" it does not update the running app

**Comment by trajano** [comment]
Digging further I found this https://github.com/nativescript-vue/nativescript-vue/issues/846 so it may likely be something specific to NativeScript-Vue

**Comment by heywhy** [comment]
HMR does work. All you need to do is to navigate to another screen then go back to the screen you just modified because the hot reload those not update a screen in place until it's rerendered and you don't necessarily need to restart the app.

**Comment by trajano** [comment]
Is that a limitation of Vue's HMR implementation?

**Comment by rigor789** [comment]
Yes with the current implementation - I'm working on improvements for Page HMR (other components that don't trigger a page to be re-rendered work fine).

This PR is part of the effort: https://github.com/NativeScript/NativeScript/pull/9460

**Comment by trajano** [comment]
@rigor789 your statement makes it sound like the issue is with NativeScript's implementation rather than VueJS' implementation.

**Comment by rigor789** [comment]
Requires changes in both NativeScript (the above PR) and nativescript-vue - the latter being much less trivial to fix.

**Comment by trajano** [comment]
@rigor789 so nothing needed from VueJS team then, that's good.

**Comment by AbrahamBrookes** [comment]
I had the same problem and the solution was to run nativescript with the `--no-hmr` flag - this skips the hmr entirely and just reboots the app on code change (found via https://stackoverflow.com/questions/59180667/nativescript-vue-not-live-updating)

`tns run android --no-hmr`

**Comment by MrSnoozles** [comment]
`no-hmr` works, but having to wait ~5-10 seconds to then navigate to the page that's being edited, unfortunately is frustrating developer experience. (I know the page can be set as start page, so the navigation part can be skipped. but still..) I soo wish there was a solution to be able to quickly style pages with NativeScript Vue.

I can imagine HMR being super complicated to get right, but it would be such a big win for Nativescript (Vue)

**Comment by trajano** [comment]
@rigor789 the PR you specified https://github.com/NativeScript/NativeScript/pull/9460 seems to be merged now.  Is the trivial fix ready?

**Comment by rigor789** [comment]
@trajano wish it was a trivial fix, though i do have a branch that is almost working, just need to revive it when i get the chance... Biggest pain point with it is preserving the navigation stack during refreshes.

**Comment by tommag21** [comment]
any update on this?

**Comment by rigor789** [comment]
> any update on this?

https://twitter.com/igor_randj/status/1599830172143788032

**Comment by tommag21** [comment]
> > any update on this?
> 
> https://twitter.com/igor_randj/status/1599830172143788032

Thank you, it seems I have a lot to catch up on. First I need to migrate my apps to Nativescript-Vue 3, are there any docs?
